rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake

```diff
@@ -70,20 +70,27 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
       "${_output_path}/${_parent_folder}/${_header_name}__rosidl_typesupport_opensplice_c.h"
       "${_output_path}/${_parent_folder}/dds_opensplice_c/${_header_name}__type_support_c.cpp")
   elseif(_extension STREQUAL ".srv")
-    list(APPEND _generated_srv_files "${_output_path}/srv/${_header_name}__rosidl_typesupport_opensplice_c.h")
-    list(APPEND _generated_srv_files "${_output_path}/srv/dds_opensplice_c/${_header_name}__type_support_c.cpp")
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    if(_parent_folder STREQUAL "srv")
+      set(_var "_generated_srv_files")
+    else()
+      message(FATAL_ERROR "Interface file with unknown parent folder: ${_idl_file}")
```

Comment:
I get an error here when generating a service in an `action` folder

```
CMake Error at /home/developer/workspaces/ros2-dev0/install/rosidl_typesupport_opensplice_c/share/rosidl_typesupport_opensplice_c/cmake/rosidl_typesupport_opensplice_c_generate_interfaces.cmake:57 (message):
  Interface file with unknown parent folder:
  /home/developer/workspaces/ros2-dev0/src/ros2/rcl_interfaces/test_msgs/action/ShaneDynamicArrayPrimitivesNested.msg
Call Stack (most recent call first):
  /home/developer/workspaces/ros2-dev0/install/ament_cmake_core/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:38 (include)
  /home/developer/workspaces/ros2-dev0/install/rosidl_cmake/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:186 (ament_execute_extensions)
  CMakeLists.txt:42 (rosidl_generate_interfaces)
```

Reply:
Oops, the error is from line 57 above. I don't think a middleware specific generator needs to check if the parent folder is known or not, so both here and above that restriction could be removed.